javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -1704,6 +1704,97 @@ public static boolean sameTree(ExpressionTree expr1, ExpressionTree expr2) {
     }
   }
 
+  /**
+   * Returns the selector expression of {@code switchExpressionTree}. For example
+   *
+   * <pre>
+   *   switch ( <em>expression</em> ) { ... }
+   * </pre>
+   *
+   * @param switchExpressionTree the switch expression whose selector expression is returned
+   * @return the selector expression of {@code switchExpressionTree}
+   */
+  public static ExpressionTree switchExpressionTreeGetExpression(Tree switchExpressionTree) {
+    try {
```

Comment:
I have an implementation suggestion that would eliminate the use of reflection.

Create a new package-private class `TreeUtils17` that contains non-reflective versions of these methods.  Have these methods in `TreeUtils` call the ones in `TreeUtils17` (possibly after a check of the JDK version number to give a more friendly error message).  `TreeUtils17` would only be compiled under Java 17.  I think we might already have conditional compilation logic like this (unless I am thinking of Daikon or Randoop; at least one of the three has it).

What do you think of this suggestion?